vpr/SRC/route/route_timing.c

```diff
@@ -65,13 +83,18 @@ struct more_sinks_than {
 #ifdef PROFILE
 static void congestion_analysis();
 static void time_on_criticality_analysis();
-constexpr int fanout_per_bin = 4;
+constexpr unsigned int fanout_per_bin = 4;
 constexpr float criticality_per_bin = 0.05;
 static vector<float> time_on_fanout;
-static vector<float> time_to_build_heap;
-static vector<int> itry_on_fanout;
+static vector<float> time_on_fanout_rebuild;
 static vector<float> time_on_criticality;
+static vector<int> itry_on_fanout;
 static vector<int> itry_on_criticality;
+static vector<int> rerouted_sinks;
+static vector<int> finished_sinks;
+static int entire_net_rerouted;
+static int entire_tree_pruned;
+static int part_tree_preserved;
 #endif

```

Comment:
I'd leave profile in; makes it a bit harder to read but could be useful in
the future.

Vaughn

On Tue, Aug 25, 2015 at 8:55 PM, Johnson Zhong <notifications@github.com>
wrote:

> In vpr/SRC/route/route_timing.c
> <https://github.com/verilog-to-routing/vtr-verilog-to-routing/pull/104#discussion_r37937379>
> :
>
> >  static vector<int> itry_on_criticality;
> > +static vector<int> rerouted_sinks;
> > +static vector<int> finished_sinks;
> > +static int entire_net_rerouted;
> > +static int entire_tree_pruned;
> > +static int part_tree_preserved;
> >  #endif
>
> I was thinking of actually taking out the code that depends on PROFILE
> being defined when I actually merge back. Should I leave them in to make
> future profiling easier (tradeoff is that the ifdef PROFILE makes the code
> harder to read)?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/verilog-to-routing/vtr-verilog-to-routing/pull/104/files#r37937379>
> .
>